tests/test_rdf.py

```diff
@@ -82,12 +95,43 @@ def test_get_ontology_terms_by_symbol():
             assert len(annotations) == 0
 
 
-def test_has_ontology_term_by_symbol():
+def test_has_ontology_term_by_symbol(test_bad_annotations):
     # Test bad annotations
-    model = load_model('test_bad_annotations')
+    model = test_bad_annotations
 
     # Get v3 from the model, as it does not have cmeta_id, to test this part of the code
     equation_graph = model.get_equation_graph()
     for variable in equation_graph:
         if str(variable) == '_c$v3':
             assert not model.has_ontology_annotation(variable, OXMETA)
+
+
+def test_get_rdf_annotation(test_simple_odes):
+    # Test rdf annotation
+    model = test_simple_odes
+
+    named_attributes = []
+    named_attrs = model.get_rdf_annotations(subject=model.rdf_identity, predicate=(PYCMLMETA, 'named-attribute'))
+    for s, p, attr in named_attrs:
+        name = model.get_rdf_value(subject=attr, predicate=(PYCMLMETA, 'name'))
+        value = model.get_rdf_value(subject=attr, predicate=(PYCMLMETA, 'value'))
+        named_attributes.append({'name': name, 'value': value})
+
+    assert len(named_attributes) == 1
+    assert named_attributes[0]['name'] == 'SuggestedForwardEulerTimestep'
+    assert named_attributes[0]['value'] == '0.0002'
+
+    # subject as string
+    named_attrib = []
+    named_attrs = model.get_rdf_annotations(subject='#test_simple_odes', predicate=(PYCMLMETA, 'named-attribute'))
+    for s, p, attr in named_attrs:
```

Comment:
Could just skip this bit and check the list equals `named_attrs` from above?

Reply:
Nope they are generators (`<generator object Graph.triples at 0x06282230>`) and are both different. If I turn them into a list the 1st one is empty (as it has already been walked through).
I could turn both into a list before walking through but don't want to give people the idea that that is somehow needed.